classes/decorator.rb

```diff
@@ -0,0 +1,24 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0...10]
+  end
```

Comment:
- [ ] As per the requirement, kindly ensure the trimmed word has a maximum of 10 characters. Your method has 11 characters right now.

Reply:
@tugsem Thanks. But it actually returns 10 characters and not 11. When `...` is used, the second number specifies the length of the substring. The second number specifies the index when `..` is used.

Reply:
👍 Great communication 💪